quickwit-storage/src/cache/disk/file.rs

```diff
@@ -0,0 +1,106 @@
+use std::array::TryFromSliceError;
+use std::time::Duration;
+
+pub type FileKey = u64;
+
+#[derive(Debug)]
+pub struct Corrupted;
+
+impl From<TryFromSliceError> for Corrupted {
+    fn from(_: TryFromSliceError) -> Self {
+        Self
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub(crate) struct FileEntry {
+    pub(crate) key: FileKey,
+    pub(crate) file_checksum: u32,
+    pub(crate) last_accessed: u64,
+    pub(crate) file_size: u64,
+}
+
+impl FileEntry {
+    pub(crate) const EMPTY: Self = Self {
+        key: 0,
+        file_checksum: 0,
+        last_accessed: 0,
+        file_size: 0,
+    };
+    pub(crate) const RAW_SIZE: usize = std::mem::size_of::<FileKey>()
+        + std::mem::size_of::<u32>()
+        + std::mem::size_of::<u64>()
+        + std::mem::size_of::<u64>()
+        + std::mem::size_of::<u32>(); // Row checksum
```

Comment:
We *could* use `std::mem::size_of::<FileEntry>()` but this could potentially change, so I think it's better to leave it as is for now so you can explicitly see the structure of the slot.

Reply:
This is an interesting comment.

Alternatively you could use `mem::size_of::<FileEntry>`  and add a unit test that will break when the size changes.

```
/// This size is used in ... . Changing it is a dangerous operation as 
/// it will break ... .
assert_eq!(RAW_SIZE, 24);
```